File: src/main/java/Log.java
import java.io.FileWriter;
import java.text.DateFormat;
import java.text.SimpleDateFormat;
import java.util.Date;

public class Log {

    public static String logFile;
    public static FileWriter fileWriter;
    private static DateFormat dateTimeFormat = new SimpleDateFormat("yyyy-MM-dd_HH:mm:ss.SSS");

    private static String time() {
        Date d = new Date();
        return dateTimeFormat.format(d);
    }

    public static void init(String logFile) {
        Log.logFile = logFile;
        try {
            fileWriter = new FileWriter(logFile);
        } catch (Exception e){
            System.out.println(e.getMessage());
        }
    }

    public static void write(String node, String msg){
        try {
            fileWriter.write(time() + " - " + node + " : " + msg + "\n");
            fileWriter.flush();
        } catch (Exception e){
            System.out.println(e.getMessage());
        }
    }

}

File: src/Log.kt
import java.io.FileWriter
import java.text.SimpleDateFormat
import java.util.*

object Log {

    private var logFile: String = ""
    private var fileWriter: FileWriter? = null
    private val dateTimeFormat = SimpleDateFormat("yyyy-MM-dd_HH:mm:ss.SSS")

    private fun time() : String = dateTimeFormat.format(Date())

    fun initialise(file: String) {
        logFile = file
        fileWriter = FileWriter(file)
    }

    fun write(node: String, msg: String) {
        if (fileWriter != null) {
            fileWriter?.write(time() + " - " + node + " : " + msg + "\n")
            fileWriter?.flush()
        }
    }

}